File: backend/internal/core/domain/integration.go
package domain

import "time"

type ProviderCredentials struct {
	ID           int       `json:"id"`
	ProjectID    string    `json:"project_id"`
	ProviderID   int       `json:"provider_id"`
	ClientID     string    `json:"client_id"`
	ClientSecret string    `json:"client_secret"`
	Scopes       []string  `json:"scopes"`
	IsActive     bool      `json:"is_active"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

File: dashboard/src/api/dto/integration.ts
export interface ProviderDisplayable {
    id: number;
    name: string;
    display_name: string;
    image_url: string;
}

export interface SaveCredsRequest {
    client_id: string;
    client_secret: string;
    project_id: string;
    provider_id: number;
    scopes: string[];
}

export interface SaveCredsResponse {
    id: string;
    created_at: string;
    updated_at: string;
    client_id: string;
    provider_id: number;
    project_id: string;
}

export type GetCredsRequest = {
    providerID: string;
    projectID: string;
};

export interface IntegrationCreds {
    id: number;
    project_id: string;
    provider_id: number;
    client_id: string;
    client_secret: string;
    scopes: string[];
    is_active: boolean;
    created_at: string;
    updated_at: string;
}

export interface ConnectedAccount {
    user_id: string
    integrations_enabled: string
    displayable_date: string
}